### PR TITLE
fix(sampling): Fix rule activation before fully backend information available - [TET-450]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Problem:**
Users who create a new project, send transactions, and quickly go to the dynamic sampling page to create a uniform rule are able to activate the rule, even if they need to update the project's SDK first and that's not what we want. This is happening because the backend uses some caching mechanisms, requiring some time for the `sdk-versions` response to bring up the correct information and not just an empty array.

**Solution:**
We have updated the frontend code, to display the `step 0`(inform your client sample rate)  and disable the uniform rule toggle for cases where the `sdk-versions` endpoint is not yet fully loaded 

**Before:**

https://user-images.githubusercontent.com/29228205/194576063-f06adf95-e713-476b-8fed-ea5cd2cadf51.mov

**After:**

https://user-images.githubusercontent.com/29228205/194576724-e2ef1652-24e9-4986-b09a-cb8d332ef405.mov

### CODE REVIEW

File: static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx [hunk 1]
```diff
@@ -125,7 +126,7 @@ export function UniformRateModal({
     );
 
     setActiveStep(
-      clientDiscard || !isProjectOnOldSDK
+      (clientDiscard || !isProjectOnOldSDK) && !isSamplingSdkVersionsBeingProcessed()
```
Comment (matejminar) [inline static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx:129]: Why do we need to check `isSamplingSdkVersionsBeingProcessed` and show the step 0 modal?

Step 0 modal should be displayed only if you don't have client outcomes (fetched from the org stats endpoint)

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx:129]: yes, I think there was a misunderstanding when we discussed this last week... will update the code in a minute.

One think I noticed is that when the "processing" is done and the project is incompatible, we don't let the users add a uniform rule, but when we don't have yet this information we do... shouldn't we remove this check only don't let users activate a uniform rule if the project is incompatible?

Comment (priscilawebdev) [reply static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx:129]: here is where we have part of the logic https://github.com/getsentry/sentry/blob/717b4e740dcc7a6a40263ce3a1fbad1bf87ea592/static/app/views/settings/project/server-side-sampling/utils/useRecommendedSdkUpgrades.tsx#L22

Comment (matejminar) [reply static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx:129]: I think we can keep it as is, that's fine 👍

File: tests/acceptance/test_project_settings_sampling.py [hunk 2]
```diff
@@ -290,7 +351,34 @@ def test_remove_specific_rule(self):
...
+                        "is_root": 1,
+                    }
+                ]
+            },
+            {
+                "data": [
+                    {
+                        "project": self.project.slug,
+                        "project_id": self.project.id,
+                        "count": 2,
```
Comment (andriisoldatenko) [inline tests/acceptance/test_project_settings_sampling.py:376]: jFYI if count is 2, previous mock should return 2 transactions.

**Review by andriisoldatenko — APPROVED** [review]

File: tests/acceptance/test_project_settings_sampling.py (hunk 2 above)
Comment (priscilawebdev) [reply tests/acceptance/test_project_settings_sampling.py:376]: Thanks for the feedback, but I think this is not relevant to this specific test, but I can update it in a follow-up.

Comment (andriisoldatenko) [reply tests/acceptance/test_project_settings_sampling.py:376]: correct, that's why i approved it already.

**Review by matejminar — APPROVED** [review]

**Pull request merged**